src/openloco/windows/ScenarioOptions.cpp

```diff
@@ -0,0 +1,1194 @@
+                            break;
+
+                        case scenario::objective_type::performance_index:
+                            *objectivePerformanceIndex = std::min<uint8_t>(*objectivePerformanceIndex + 5, scenario::max_objective_performance_index);
+                            break;
+
+                        case scenario::objective_type::cargo_delivery:
+                        {
+                            uint32_t newDeliveredCargoAmount = *objectiveDeliveredCargoAmount;
+                            newDeliveredCargoAmount += 100;
+                            uint16_t divisor = 100;
+
+                            if (*_clickRepeatTicks >= 100)
+                            {
+                                newDeliveredCargoAmount += 900;
+                                divisor = 1000;
+                            }
+                            if (*_clickRepeatTicks >= 200)
+                            {
+                                newDeliveredCargoAmount += 9000;
+                                divisor = 10000;
+                            }
+
+                            // Round it off to the nearest multiple of the divisor.
+                            uint16_t quotient = newDeliveredCargoAmount / divisor;
+                            newDeliveredCargoAmount = quotient * divisor;
+
+                            *objectiveDeliveredCargoAmount = std::min<uint32_t>(newDeliveredCargoAmount, scenario::max_objective_delivered_cargo);
+                            break;
+                        }
+                    }
+
+                    self->invalidate();
+                    break;
+                }
+
+                case widx::objective_cargo_btn:
+                {
+                    const uint8_t maxCargoObjects = static_cast<uint8_t>(objectmgr::get_max_objects(object_type::cargo));
+                    uint16_t numCargoObjects = 0;
+                    for (uint16_t cargoIdx = 0; cargoIdx < maxCargoObjects; cargoIdx++)
+                    {
+                        auto cargoObject = objectmgr::get<cargo_object>(cargoIdx);
+                        if (cargoObject != nullptr)
+                            numCargoObjects++;
+                    }
+
+                    widget_t& target = self->widgets[widx::objective_cargo];
+                    dropdown::show(self->x + target.left, self->y + target.top, target.width() - 3, target.height(), self->colours[1], numCargoObjects, 0x80);
+
+                    uint16_t ddIdx = 0;
+                    for (uint16_t cargoIdx = 0; cargoIdx < maxCargoObjects; cargoIdx++)
+                    {
+                        auto cargoObject = objectmgr::get<cargo_object>(cargoIdx);
+                        if (cargoObject == nullptr)
+                            continue;
+
+                        dropdown::add(ddIdx, string_ids::dropdown_stringid, { cargoObject->name, cargoIdx });
```

Comment:
```suggestion
                        dropdown::add(ddIdx, string_ids::dropdown_stringid, cargoObject->name);
```

Clean up when removing the dropdown argument accessor